nomad/state/state_store.go

```diff
@@ -833,28 +856,26 @@ func (s *StateStore) UpdateNodeStatus(index uint64, nodeID, status string, updat
 
 	// Add the event if given
 	if event != nil {
-		appendNodeEvents(index, copyNode, []*structs.NodeEvent{event})
+		appendNodeEvents(txn.Index, copyNode, []*structs.NodeEvent{event})
 	}
 
 	// Update the status in the copy
 	copyNode.Status = status
-	copyNode.ModifyIndex = index
+	copyNode.ModifyIndex = txn.Index
 
 	// Insert the node
 	if err := txn.Insert("nodes", copyNode); err != nil {
 		return fmt.Errorf("node update failed: %v", err)
 	}
-	if err := txn.Insert("index", &IndexEntry{"nodes", index}); err != nil {
+	if err := txn.Insert("index", &IndexEntry{"nodes", txn.Index}); err != nil {
 		return fmt.Errorf("index update failed: %v", err)
 	}
-
-	txn.Commit()
```

Comment:
why doesn't this commit here anymore?

Reply:
sketchy github diff, UpdateNodeStatus initializes the transaction, calls updateNodeStatusTxn to do the logic, and returns an error, the transactions committed in UpdateNodeStatus